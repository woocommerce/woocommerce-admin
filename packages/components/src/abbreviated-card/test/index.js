/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { createElement } from '@wordpress/element';

/**
 * Internal dependencies
 */
import AbbreviatedCard from '../';

describe( 'AbbreviatedCard', () => {
	test( 'it renders correctly', () => {
		const { container, getByText } = render(
			<AbbreviatedCard icon={ <div>icon</div> } href="#">
				<p>Abbreviated card content</p>
			</AbbreviatedCard>
		);
		expect( container ).toMatchSnapshot();

		// should have correct content
		expect( getByText( 'Abbreviated card content' ) ).toBeInTheDocument();

		// should have correct class
		expect(
			container.getElementsByClassName( 'woocommerce-abbreviated-card' )
		).toHaveLength( 1 );
	} );
} );
