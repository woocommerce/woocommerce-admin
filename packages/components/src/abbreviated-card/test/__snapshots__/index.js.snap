// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbbreviatedCard it renders correctly 1`] = `
<div>
  <div
    class="components-surface components-card woocommerce-abbreviated-card css-1vyvcpq-View-Surface-getBorders-primary-Card-rounded em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Card"
  >
    <div
      class="css-mgwsf4-View-Content em57xhy0"
    >
      <div
        class="components-card__body components-card-body css-1sfrl79-View-Body-borderRadius em57xhy0"
        data-wp-c16t="true"
        data-wp-component="CardBody"
      >
        <a
          data-link-type="wc-admin"
          href="#"
        >
          <div
            class="woocommerce-abbreviated-card__icon"
          >
            <div
              size="24"
            >
              icon
            </div>
          </div>
          <div
            class="woocommerce-abbreviated-card__content"
          >
            <p>
              Abbreviated card content
            </p>
          </div>
        </a>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Elevation"
    />
    <div
      aria-hidden="true"
      class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Elevation"
    />
  </div>
</div>
`;
